### PR TITLE
CM-105: fetch when default filters change

### PR DESCRIPTION
**Pull request by jdolkowski**
https://openimis.atlassian.net/browse/CM-105

When using cache default filters were not fetched once changed. Changing default filters fill result in filtering.

### CODE REVIEW

**Review by sniedzielski — CHANGES_REQUESTED** [review]
I requested change because I'm afraid that this solution could break custom filters implementation provided in CM-71. We have to think how to integrate our features to have both working as expected.

File: src/components/generics/Searcher.js [hunk 1]
```diff
@@ -205,6 +206,23 @@ class Searcher extends Component {
     );
   }
 
+  componentDidUpdate(prevProps, prevState, snapshot) {
```
[the remark this answers is not in the document]
Comment (sniedzielski) [reply src/components/generics/Searcher.js:209]: Peek 2023-06-14 17-02
@jdolkowski This is how it works CM-71 after integrating with this change from core and social protection CM-71 with  CM-105

[the remark this answers is not in the document]
Comment (jdolkowski) [reply src/components/generics/Searcher.js:212]: now it should work fine, it also caches filters for each tab

[the remark this answers is not in the document]
Comment (sniedzielski) [reply src/components/generics/Searcher.js:212]: thx

**Pull request merged**